surrogates/common_utils/doc_string.py

```diff
@@ -0,0 +1,141 @@
+##==============================================================================
+## BHPTNRSurrogate module
+## Description : docs for different surrogate model wrappers
+## Author : Tousif Islam, Nov 2022 [tislam@umassd.edu / tousifislam24@gmail.com]
+##==============================================================================
+
+import inspect
+
+def copy_doc(base_func, model_func):
+    """
+    function to copy doc string from the base function and the model function
+    and join them to create docstring fot the target function
+    """
+    def wrapper(target_func):
+        target_func.__doc__ = base_func.__doc__ + model_func.__doc__
+        return target_func
+    return wrapper
+    
+    
+def generic_doc_for_models() -> None:
+    """
+    ## -------------------------------------------------------------------------- ##
+    ## general overview
+    ## -------------------------------------------------------------------------- ##
+    
+    Description : warpper to generate BHPT surrogate waveforms
+    
+    Input
+    =====
+    q:      mass ratio
+    
+    spin1: spin of the primary black hole / Default: None
+           Not Implemented in most models
+    
+    spin2: spin of the secondary black hole / Default: None
+           Not Implemented in any model so far
+    
+    ecc: eccentricity / Default: None
+         Not Implemented in any model so far
+    
+    ano: mean anomaly / Default: None
+         Not Implemented in any model so far
+    
+    modes:  list of modes
+            Default (None) corresponds to all available modes in the model
+
+            e.g. [(2,2),(2,1),(3,1),(3,2),(3,3),(4,2),(4,3),(4,4),
+                  (5,3),(5,4),(5,5),(6,4),(6,5),(6,6),(7,5),(7,6),
+                  (7,7),(8,6),(8,7),(8,8),(9,7),(9,8),(9,9),(10,8),(10,9)]
+             for BHPTNRSur1dq1e4
+            
+    M_tot:     total mass of the binary in solar masses
+               Default: None (in which case a geometric waveform is returned)
+    
+    dist_mpc:  distance of the binary from the observer in Mpc
+               Default: None (in which case geometric wf is returned)
+               
+    orb_phase: orbital phase
+    
+    inclination: inclination angle
+    
+    lmax:  5 (default)
+           Modes upto l=5 are NR calibrated. Modes l>6 are uncalibrated
```

Comment:
l=5 is a choice for the current model, right? Might be better to remind the user that (i) modes are only calibrated to NR up to some maximum value of ell, and (ii) consult a specific model's documentation for information on how modes are calibrated to NR.

Reply:
Good point. Made the change.